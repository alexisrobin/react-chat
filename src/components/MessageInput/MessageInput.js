import React, { Component } from 'react';
import './MessageInput.scss';
import { Button, Container, Col, Row } from 'reactstrap';
import { LocalForm, Control, actions } from 'react-redux-form';

const required = val => val && val.length;

/**
 * Chat's messages input.
 */
class MessageInput extends Component {

    /**
     * Handle message's submission.
     * Reset message's form.
     */
    handleSubmit = values => {
        this.props.addMessage(values.message, this.props.currentUserId);
        this.formDispatch(actions.reset('local'));
    }

    render() {
        return (
            <Container className="p-0">
                <LocalForm
                    onSubmit={(values) => this.handleSubmit(values)}
                    getDispatch={(dispatch) => this.formDispatch = dispatch}>
                    <Row className="form-group">
                        <Col xs={10}>
                            <Control.textarea
                                model=".message"
                                className="form-control message-textarea"
                                color="secondary"
                                validators={{
                                    required
                                }} />
                        </Col>
                        <Col xs={2}>
                                <Button outline type="submit" color="secondary" className="w-100 h-100">Send</Button>
                            </Col>
                    </Row>
                </LocalForm>
            </Container>
                )
            }
        }
        
export default MessageInput;