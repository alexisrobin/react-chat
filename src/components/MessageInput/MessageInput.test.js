import React from 'react';
import { expect } from 'chai';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import MessageInput from './MessageInput';
import { Control, LocalForm } from 'react-redux-form';

describe('<MessageInput />', () => {
    it('should match its reference snapshot', () => {
        const wrapper = shallow(
            <MessageInput />
        );

        expect(wrapper).to.matchSnapshot()
    })

    it('should trigger its `addMessage` prop', () => {
        const message = "testing message";
        const addMessage = sinon.spy()
        const wrapper = mount(
            <MessageInput 
                addMessage={addMessage}
                currentUserId={1}/>
        )
        wrapper.find(Control.textarea).simulate('change', {target: {value: message}});
        wrapper.find(LocalForm).simulate('submit');
        expect(addMessage).to.have.been.calledWith(message, 1);
      })
})