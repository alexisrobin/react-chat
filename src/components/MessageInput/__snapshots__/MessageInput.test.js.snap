// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageInput /> should match its reference snapshot 1`] = `
<Container
  className="p-1"
  tag="div"
>
  <LocalForm
    getDispatch={[Function]}
    initialState={Object {}}
    model="local"
    onSubmit={[Function]}
  >
    <Row
      className="form-group"
      tag="div"
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs={9}
      >
        <DefaultConnectedControlTextArea
          className="form-control message-textarea"
          color="secondary"
          model=".message"
          validators={
            Object {
              "required": [Function],
            }
          }
        />
      </Col>
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs={3}
      >
        <Button
          className="w-100 h-100"
          color="secondary"
          outline={true}
          tag="button"
          type="submit"
        >
          <span
            className="fa fa-rocket fa-lg"
          />
        </Button>
      </Col>
    </Row>
  </LocalForm>
</Container>
`;
